### ISSUE TITLE
mvs-texturing gives empty results

### ISSUE THREAD

**Issue by mojovski** [nmoehrle/mvs-texturing#11]
After running mvs-texturing I end up with [empty results, shared here](https://drive.google.com/open?id=0B2fi7-kZoDhPOXhXbmJmQzBrZGs&authuser=0)
The output of the console is
```
Initialized 1360 views (max ID is 1359), took 47ms.
        Loading 100%... done. (Took 0.513s)                                     
        Generating validity masks 100%... done. (Took 0.086s)                   
        Calculating gradient magnitude 100%... done. (Took 17.225s)             
Building adjacency graph: 
        Adding edges 100%... done. (Took 0.087s)                                
	50399 total edges.
Building MRF graph:
        Building collision model for visibility tests 100%... done. (Took 0.049s)
        Calculating face qualities 100%... done. (Took 12.263s)                 
	Maximum quality of a face within an image: 0
	Clamping qualities to 0 within discretization.
	Writing data cost file... done.
	100798 faces have not been seen by a view.
Running MRF optimization:
	Iteration	Energy	Runtime
	0	1007980000000	0.002
	1	1007980000000
	Converged
	Took: 0.795s
Generating texture patches:
	Running... done. (Took 0.093s)
	0 texture patches.
Running global seam leveling:
	Create matrices for optimization...  done.
	Lhs dimensionality: 201596 x 201596
Calculating adjustments:
		Color channel 0: CG took 0 iterations. Residual is 0
		Color channel 2: CG took 0 iterations. Residual is 0
		Color channel 1: CG took 0 iterations. Residual is 0
		Took 0.168 seconds
Running local seam leveling:
Building objmodel:
	Sorting texture patches... done.
	Generating texture atlases (working on atlas 0) 100%... done.
	Filling invalid pixels of the texture atlases... done. (Took: 0s)
	Writing files... done.
Whole texturing procedure took: 15.597s

```

It looks to me that the texture atlas is empty. :)
Is there a way how I can debug this issue?

Thanks alot!

**Comment by nmoehrle** [comment]
It appears that none of the faces of your input mesh has been seen by a view (I assume you have around 100k faces):
```100798 faces have not been seen by a view.```
To debug the issue please verify that the camera parameters are correct:
- open the scene and the mesh in UMVE check if the camera frustas look towards the view.
- check your parameters against the specification in the help text (```./texrecon ```)

I am sorry that it took me so long to respond, because of earlier issues I didn't respond to the email but in the Github Android App - didn't came through either... Don't know what's wrong there...

**Comment by mojovski** [comment]
Thank you for the hint with umve. Its indeed a great tool! :)

1. As for the camera orientation, I had indeed a wrong transformation (didnt see, that it must be a cam2world transformation). Now it looks correct.
![Camera Frustum Check](http://ibin.co/23w96pguLbM0)

As for the parameters specification in the help text of texrecon, I have not found any hints.
Just to be sure, could you confirm the following information in the mve file?

A view contains in my case only a pinhole camera model (view.set_camera(...) ) with
* distortion  dist[0]=dist[1]=0
* focal length: in **mm** (currently 383)
* principal point: center of the image
* pose: cam2world rotation matrix (row based) and the translation.
* An pinhole image (undistorted), set with view.set_image(...).


Can you recommend me where to look further? 


Thank you very much for your support!

**Comment by mojovski** [comment]
I think I got it! :)
The issue was, that I passed the focal length and the principal points unnormalized. However, MVE expects these parameters to be normalized by the longest edge in the image (e=max(width, height)).
Then, the principal point is passed as (0.5, 0.5) and the focal length as f_metric/e.

When visualizing the mve scene in umve, the camera frusta should look like this (and not as has been shown in the previous image).
![correct camera frusta](http://ibin.co/23wZmL6v1A8m)

Unfortunately, there is another issue with the resulting mesh (many previous mesh elements are missing). But this is another story.

Nils, than you very much for your help!
